common/src/main/java/com/microsoft/identity/common/internal/controllers/ExceptionAdapter.java

```diff
@@ -99,29 +99,19 @@ public static BaseException exceptionFromAcquireTokenResult(final AcquireTokenRe
             );
         }
 
-        final TokenResult tokenResult = result.getTokenResult();
+        return exceptionFromTokenResult(result.getTokenResult());
+    }
+
+    /**
+     * Extract exception from a given TokenResult object.
+     * */
+    public static ServiceException exceptionFromTokenResult(TokenResult tokenResult) {
+        final String methodName = ":exceptionFromTokenResult";
         final TokenErrorResponse tokenErrorResponse;
 
         if (tokenResult != null && !tokenResult.getSuccess()) {
             tokenErrorResponse = tokenResult.getErrorResponse();
 
-            if (tokenErrorResponse.getError().equalsIgnoreCase(AuthenticationConstants.OAuth2ErrorCode.INVALID_GRANT)) {
```

Comment:
@rpdome As synced offline, this can be added back which means no changes will be required in Msal or ad-accounts controllers. 

In addition any parsing to look and map for UIRequiredException can be done here, like looking of "interaction_required" here as discussed offline